bugsnag-android-core/src/main/java/com/bugsnag/android/internal/ImmutableConfig.kt

```diff
@@ -193,6 +222,8 @@ internal fun sanitiseConfiguration(
         packageManager.getApplicationInfo(packageName, PackageManager.GET_META_DATA)
     }.getOrNull()
 
+    validateApiKey(configuration.apiKey, configuration.logger!!)
```

Comment:
@lemnik I am not sure about this part. Since the apiKey does need the validation check so the configuration.logger can not be null. But this might throw an exception.